### PR TITLE
Super Admin lurker capability WIP (frontend)

### PR DESCRIPTION
**Pull request by jessykate**
`fetchMyMemberships` is the main method that determined if a group can be viewed or not. it is called in two places, on the `boot` page and the `group-setup-page`. this work modifies the `fetchMyMemberships` method to accept `groupId` as an argument. it then modifies the `boot` page and `group-setup-page` to use this functionality to pass the group id being requested to the backend. it also updates a couple of methods on the user model. all this is done so that we can pass the requested group's id to the backend, and create a fake membership in that group when it is being requested by a superadmin.

right now this works for viewing a specific group page, but there's two issues: 

1. it doesn't work for a second group unless you do a hard refresh of the page (because only the hard refresh re-sends the group id to the backend)
2. if you view a bucket page instead of a group page, we need a second step to get the group id from the bucket id, but this isn't working currently and i'm not sure exactly why. i'll point this out in an inline code comment.

### CODE REVIEW

File: app/boot.coffee [hunk 1]
```diff
@@ -8,7 +8,22 @@ global.cobudgetApp.run ($auth, CurrentUser, Dialog, LoadBar, $location, $q, Reco
...
+    if path_components[1] == "groups"
+      groupId = path_components[2]
+    else if path_components[1] == "buckets"
+      bucketId = parseInt path_components[2]
+      console.log("bucketId")
+      console.log(bucketId)
+      Records.buckets.findOrFetchById(bucketId).then(bucket) ->
+        console.log("bucket")
+        console.log(bucket)
+        groupId = bucket.group().id
```
Comment (jessykate) [inline app/boot.coffee]: this section starting on line 18 with `Records.buckets.findOrFetchById(bucketId).then(bucket)` isn't getting executed, but i am not sure why. perhaps you can't have a promise inside another promise? @jlopker i am guessing you might know what is happening here better as i am just poking around in the dark and making shit up basically :).

File: app/models/user-model.coffee [hunk 2]
```diff
@@ -36,12 +36,12 @@ global.cobudgetApp.factory 'UserModel', (BaseModel) ->
...
-      !!_.find @memberships(), (membership) ->
-        membership.groupId == group.id
+      @isSuperAdmin || (!!_.find @memberships(), (membership) ->
+        membership.groupId == group.id)
 
     isAdminOf: (group) ->
-      !!_.find @memberships(), (membership) ->
-        membership.groupId == group.id && membership.isAdmin
+      @isSuperAdmin || (!!_.find @memberships(), (membership) ->
+        membership.groupId == group.id && membership.isAdmin)
```
Comment (jessykate) [inline app/models/user-model.coffee]: TBH I am not sure 100% sure if these are needed. I added them before realizing that we needed to actually fake the membership to get the group data from the backend. so perhaps we can get rid of this

File: app/services/user-can.coffee [hunk 3]
```diff
@@ -8,7 +8,7 @@ global.cobudgetApp.factory 'UserCan', (CurrentUser, $location, $q, Records, Toas
         groupId: group.id,
         memberId: global.cobudgetApp.currentUserId
       })
-      validMemberships.length == 1
+      CurrentUser.isSuperAdmin || validMemberships.length == 1
```
Comment (jessykate) [inline app/services/user-can.coffee]: same comment here as one `user-model.coffee`

File: app/boot.coffee [hunk 4]
```diff
@@ -11,6 +11,17 @@ global.cobudgetApp.run ($auth, CurrentUser, Dialog, LoadBar, $location, $q, Reco
...
     global.cobudgetApp.currentUserId = user.id
+    if user.is_super_admin
+      pathComponents = $location.path().split('/')
+      if pathComponents[1] == "groups"
+        groupId = pathComponents[2]
+        Records.memberships.fetchOneMembership(groupId).then (data) ->
+          membershipsLoadedDeferred.resolve(data)
+      else if pathComponents[1] == "buckets"
+        bucketId = parseInt pathComponents[2]
+        Records.buckets.findOrFetchById(bucketId).then (bucket) ->
```
Comment (jlopker) [inline app/boot.coffee]: add a new backend route that takes in a bucket id and returns a fake membership

**Comment by chime-mu** [comment]
@jlopker Tested and seems to work in general, except when going to the root URL (`localhost:9000` when running locally) then it goes into what looks like an infinite loop.

**Comment by jlopker** [comment]
good catch! fixed.

**Pull request merged**